driver-core/src/test/unit/com/mongodb/client/model/AggregatesSpecification.groovy

```diff
@@ -544,6 +554,99 @@ class AggregatesSpecification extends Specification {
         }''')
     }
 
+    def 'should render $search'() {
+        when:
+        BsonDocument searchDoc = toBson(
+                search(
+                        (SearchOperator) exists(fieldPath('fieldName')),
```

Comment:
Is this casting needed due to Groovy weirdness?

Reply:
Yes, this is due to how Groovy dispatches methods. Our implementations of `SearchOperator` and `SearchCollector` has the same run-time type, which makes both `Aggregates.search` methods valid candidates, and the dispatching code fails to pick the right one, because it completely disregards the compile-time types of the arguments passed to a method. Yet, it does not disregard the compile-time type of the argument when an explicit cast is added... This language is insane.